### ISSUE TITLE
Lightdm seems to reset audio volume of system (pulse and pipewire)

### ISSUE THREAD

**Issue by Johndeep** [canonical/lightdm#213]
What happens (at least in my case):
- If I log out and log in via lightdm, the active audio volume (run by pipewire) seems to be enabled and set at very odd 76%. Microphone is also always switched on and set at odd but always consistent volume. 

Steps to reproduce:
- Being on Debian Unstable (optionally Xfce4 desktop ver. 4.16.1-1), have lightdm (ver. 1.26.0-7)
- Configured pipewire and pipewire-pulse (so system is basically running pipewire ver. 0.3.38-2), masking pulseaudio
- Use light-locker (ver. 1.8.0-3) to lock system. Or, just log out and log in again.

Various attempts:
- I've tried to isolate the problems by removing all configurations from other parts (alsa, pipewire, pulse, etc.). And it boils down to lightdm it seems. If I try to lock my screen via xfce4-screensaver, issue does not happen.
- A normal reboot/restart of system with initial login (via lightdm) does not trigger issue, volume stays the same as before.

I might be wrong about lightdm, but it is quite unsettling and strange the volume always resets after lightdm locks the screen.

**Comment by Jasmin68k** [comment]
I am also using Debian Unstable.

The same issue started to happen to me after unattended-upgrades removed pipewire-media-session and installed pipewire-pulse, wireplumber and libwireplumber-0.4-0.

It can be solved by either switching to another display manager, I tried xdm, or by removing pipewire-pulse, wireplumber and libwireplumber-0.4-0 and reinstalling pipewire-media-session.

**Comment by Johndeep** [comment]
If that is the case, then especially _pipewire-pulse_ seems to have some interaction problems with _lightdm_. Or somehow _lightdm_ boldly manipulates the audio volume for whatever reason.

But unfortunately I don't want to remove _pipewire-pulse_ yet, since it provides me with very handy Bluetooth Headset abilities I don't want to miss out. In fact, Bluetooth Headset ability (easy switch between pure audio and mix of audio and microphone) is the reason why I ditched _pulseaudio_ in the first place.

Furthermore, _pipewire-media-session_ seems to become deprecated in the future, so I would rather use _wireplumper_ instead. But this does not matter, because I also had _pipewire-media-session_ installed alongside with _pipewire-pulse_, which still had the audio volume problem nonetheless.

My workaround solution (**not the solution** for this issue still) is to purge or remove _light-locker_ and install _xfce4-screensaver_ instead, for xfce4 desktop at least. This not only allow any Bluetooth connection still be active even after the screen is locked, but it also prevents any unwanted audio volume manipulation.
The one caveat is that if you log out and log in again, which will call up the login dialog of _lightdm_ instead, audio volume will be manipulated by _lightdm_ again..

So I would agree to searching for another session manager besides _lightdm_ to work around this problem. But it would not solve the actual interaction problem between _lightdm_ and _pipewire-pulse_.

**Comment by Jasmin68k** [comment]
I also tried pipewire-media-session + pipewire-pulse having the same problem.

I agree with using wireplumber, since pipewire-media-session will become deprecated, and just using another display manager.

**Comment by Johndeep** [comment]
Okay, one reason less that this problem is more global than I thought is after I've reinstalled Debian again, but still without pipewire. Currently, it runs Debian Testing (Dec. 2021) with pure pulseaudio only.
The exact same problem is still happening with how after every reboot of the system, the volume is set to a fixed number again.

Since this problem is not only affecting pipewire, but also pulseaudio and thus probably the volume management in general, I'll change the topic name.

**Comment by Johndeep** [comment]
After re-checking the logs and privileges, it seems that **this can happen due of too much privilege**.

To explain this, for a normal user using a Desktop, it seems to be recommended to have as few privileges as possible, the group **sudo** should suffice at most. Since I've installed my system with Debian, it seems that **Debian always gave me too much privileges as default**, which also includes the **audio group**.

To work around this strange arbitrary audio volume change as I use lightdm to lock the screen or just log out, I've removed myself from the audio group. In fact, I removed myself from all groups except for sudo group.

To remove a user **from a certain group** (i.e. the audio group):
`# gpasswd -d <user> audio`

To remove a user from **all groups except for a certain group** (i.e. sudo group):
`# usermod -G sudo <user>`

Now even after a logout or screen lock via light-locker, the volume settings are left untouched! Even though this is more of a workaround, I'll close this issue as solved.

The question is now, why does the user (with audio group privileges) always set the volume when locking the screen or logging out? Might this be caused by the interaction between the audio server and lightdm?
